File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    answer = 0
    def dfs(network, start, visited):
        visited[start] = True
        for i in range(n):
            if i != start and network[start][i] == 1 and not visited[i]:
                dfs(network, i, visited)

    visited = [False]*n
    for i in range(n):
        if not visited[i]:
            answer += 1
            dfs(computers, i, visited)

    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    public int solution(int n, int[][] computers) {
        int answer = 0;
        
        boolean[] visited = new boolean[n];
        for (int i = 0; i < n; i++) {
            if (!visited[i]) {
                dfs(computers, i, visited);
                answer++;
            }
        }
        
        return answer;
    }
    
    public void dfs(int[][] grid, int start, boolean[] visited) {
        visited[start] = true;
        
        int[] nxts = grid[start];
        for (int i = 0; i < nxts.length; i++) {
            if (!visited[i] && nxts[i] == 1) {
                dfs(grid, i, visited);
            }
        }
    }
}